### ISSUE TITLE
AdamW algorithm not matched with Pytorch

### ISSUE THREAD

**Issue by Kokaine** [linhaowei1/NLPDL-2025Fall#13]
The definition given in HW1-4.3 Algorithm 1 AdamW Optimizer is not the definition used by Pytorch 2.9 documentation. The PyTorch implementation applies weight decay first. However, in this assignment, the instruction shows that the weight decay is used after updating the parameters. Pytest couldn't pass based on the instructions.

Assignment:
<img width="771" height="376" alt="Image" src="https://github.com/user-attachments/assets/221261e6-8cc4-4352-a08a-1125b38d5426" />

PyTorch:
<img width="1169" height="699" alt="Image" src="https://github.com/user-attachments/assets/f8ad89da-5a0c-4413-b7d4-b52164b9d4b2" />

**Comment by WangXiangyv** [comment]
Thank you for pointing out. We have noticed the divergence between the pseudo code we have provided and the PyTorch implementation. However, we validated that our code can pass the test before releasing the lab. Hence I suspect that there is something else wrong in your implementation. You could try to put the weight decay step in the front of your code and check if the test can be passed to further determine if this is the key. And we will do some further tests in the meanwhile. You can pay close attention to this issue in 1~2 days.

**Comment by WangXiangyv** [comment]
We will refine the test content to ensure that impl that precisely follows the pseudo code can pass. If your code can pass the current test, the outcome will not be affected.
